### PR TITLE
Add scheduled jobs support ⏰

### PR DESCRIPTION
**Pull request by dmitrychopey**

### CODE REVIEW

**Comment by sailor** [comment]
think you'll need to rebase 😉

**Comment by sailor** [comment]
+ if you could upgrade the protocol-redis gem to 0.3.1 which contains a bugfix for LREM it could be cool. I'm waiting for the merge of this one as well https://github.com/socketry/async-redis/pull/26

File: lib/quiq/scheduler.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    def start
+      Async do
+        loop do
+          result = Quiq.redis.zrange(SET_NAME, 0, 0, withscores: true)
+          payload, scheduled_at = result
+
+          if payload && scheduled_at.to_f < Time.now.to_f
+            job = JSON.parse(payload)
+
+            #TODO: Imolement fine-grained locking
```
Comment (sailor) [inline lib/quiq/scheduler.rb]:
```suggestion
            # TODO: Implement fine-grained locking
```

File: lib/quiq/client.rb [hunk 2]
```diff
@@ -5,15 +5,20 @@
 
 module Quiq
   class Client
-    def push(job)
+    def push(job, scheduled_at)
       serialized = JSON.dump(job.serialize)
+
       Async do
-        Quiq.redis.lpush("queue:#{job.queue_name}", serialized)
+        if scheduled_at
```
Comment (airclovis) [inline lib/quiq/client.rb]: I think it should be better to test if scheduled_at > now() here. Wdyt ?

Comment (dmitrychopey) [reply lib/quiq/client.rb]: In this case it will never process some jobs that were scheduled while server was down.

**Pull request merged**